### ISSUE TITLE
[Feature] Auto-adding Donations from Streamtip.com

### ISSUE THREAD

**Issue by Gocnak** [Gocnak/Botnak#11]
Self explanatory. Add a slot in the new Settings GUI for the API key of the site, and have a heartbeat thread check every 5 or so seconds for a new donation.

Sort by IDs of the transaction, for when the first transaction is updated, and is a new ID of donation, update the user and the hashmap.

**Comment by Triscy** [comment]
I would very much appreciate support for http://www.twitchalerts.com/ as well, as that's the site that I use at the moment.

**Comment by Gocnak** [comment]
Does that even have an API? If it doesn't, I can't implement it.

**Comment by Decicus** [comment]
It does have an API. However, you need to send an email to api@twitchalerts.com to use it apparently.

**Comment by Triscy** [comment]
Yea; it does have an API. What Decicus said.

**Comment by Gocnak** [comment]
Alright I see. But there's something that I've realized.

I've added steamtip support into Botnak because streamtip itself didn't have any notification software I wanted. Then TNotifier came out, but its features are just things I don't need now.

Furthermore, seeing twitchalerts now makes me really contemplate why I'm adding the support in the first place. If I add support from it, I'm essentially just doing what the website already does for you. It notifies you of followers, donations, and subscribers. Unless you want Botnak to say something in direct response to one of those events, there's no point for me to add support, even for streamtip.

I mean, I've already done streamtip and subscriber support so I either cut out the last month of work, or just push it as is, and people decide whether to use it or not.

I guess I just need enlightenment on why I should continue adding support for sites like these.

I wanted streamtip support so I didn't have another program running, and I wanted the notification to be discrete (part of chat). People want me to add exactly what twitchalerts and tnotifier already do to Botnak, and I've just reached the point of "why?" 

It's nothing personal on you guys, it's just curiosity to see what the interest is for.
